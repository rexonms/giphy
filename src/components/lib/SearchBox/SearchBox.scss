.container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  text-align: center;
  min-width: 70vw;
  input {
    flex: 3;
    font-size: $fontSizeSmall;
    outline: 0;
    transition: background-color 200ms linear;
    padding: 18px;
    margin: 0;
    background-color: $greyE;
  }
  button {
    flex: 1;
    outline: 0;
    margin: auto;
    padding:$defaultPadding;
    background: $grey2;
    border: none;
    font-size: $fontSizeDefault;
    transition: background-color .2s linear;
    color: $white;
  }
  button:hover {
    background: $grey3;
    transition: background-color .2s linear;
  }
}

