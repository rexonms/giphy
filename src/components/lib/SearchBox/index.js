import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Styles from './SearchBox.scss';

const propTypes = {
  /** Default value that will to be displayed if available */
  value: PropTypes.string,
  /** Placeholder for the input box */
  placeholder: PropTypes.string.isRequired,
  /** Search button text */
  buttonText: PropTypes.string.isRequired,
  /** Action that passes the value */
  onSearchHandler: PropTypes.func.isRequired,
};
const defaultProps = {
  value: '',
};

class SearchBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value,
    };
    this.onChangeHandler = this.onChangeHandler.bind(this);
    this.onClickHandler = this.onClickHandler.bind(this);
  }
  onChangeHandler(e) {
    this.setState({
      value: e.target.value,
    });
  }
  onClickHandler(e) {
    e.preventDefault();
    if (this.isValid()) {
      this.props.onSearchHandler(this.state.value);
      this.setState({ value: '' });
    }
  }
  isValid() {
    return this.state.value.length > 1;
  }
  render() {
    return (
      <form onSubmit={e => this.onChangeHandler(e)} className={Styles.container}>
        <input
          name={this.props.value}
          value={this.state.value}
          placeholder={this.props.placeholder}
          onChange={e => this.onChangeHandler(e)}
        />
        <button onClick={e => this.onClickHandler(e)}>{this.props.buttonText}</button>
      </form>
    );
  }
}

SearchBox.propTypes = propTypes;
SearchBox.defaultProps = defaultProps;

export default SearchBox;
