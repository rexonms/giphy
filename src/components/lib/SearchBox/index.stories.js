import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';

import SearchBox from './index';

const stories = storiesOf('lib/SearchBox', module);
export const props = {
  value: '',
  buttonText: 'Search',
  placeholder: 'Start Typing',
  onSearchHandler: action('Search'),
};

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('SearchBox', withInfo('Search Input Box')(() => <SearchBox {...props} />));
