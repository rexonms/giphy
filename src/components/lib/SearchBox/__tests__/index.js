import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from './../index';

describe('SearchBox', () => {
  let props;
  let shallowSearchBox;
  const searchBox = () => {
    if (!shallowSearchBox) {
      shallowSearchBox = shallow(<SearchBox {...props} />);
    }
    return shallowSearchBox;
  };
  beforeEach(() => {
    props = {
      placeholder: 'placeholder',
      value: 'value',
      buttonText: 'Button Text',
      onSearchHandler: jest.fn(),
    };
    shallowSearchBox = undefined;
  });
  it('should match previous snapshot', () => {
    expect(searchBox()).toMatchSnapshot();
  });
});
