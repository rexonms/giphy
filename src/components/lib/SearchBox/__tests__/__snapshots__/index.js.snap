// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should match previous snapshot 1`] = `
<form
  className="container"
  onSubmit={[Function]}
>
  <input
    name="value"
    onChange={[Function]}
    placeholder="placeholder"
    value="value"
  />
  <button
    onClick={[Function]}
  >
    Button Text
  </button>
</form>
`;
